calliope/test/test_constraint_results.py

```diff
@@ -150,8 +150,38 @@ def override(feasibility, cap_val):
 
 
 class TestGroupConstraints:
+
+    def test_no_group_constraint(self):
+        model = build_model(model_file="group_constraints.yaml")
+        model.run()
+        expensive_generation = (model.get_formatted_array("carrier_prod")
+                                     .to_dataframe()
+                                     .reset_index()
+                                     .groupby("techs")
+                                     .carrier_prod
+                                     .sum()
+                                     .loc["expensive_supply"])
+        assert expensive_generation == 0
+
+    def test_switched_off_group_constraint(self):
```

Comment:
If using an override dict instead of a new model definition:
```
model = build_model(
           override_dict={'group_constraints.example_demand_share_max_constraint.exists': False},
            model_file="demand_share.yaml",
            scenario="demand_share_max_systemwide"
        )
```